### PR TITLE
Add rewrites for new internal BV operators

### PR DESCRIPTION
**Pull request by ajreynol**
Changes taken from proof-new.

This does not impact rewriting for any core BV operator.

FYI @vleni

### CODE REVIEW

**Review by mpreiner — APPROVED** [review]

File: src/theory/bv/theory_bv_rewrite_rules.h [hunk 1]
```diff
@@ -81,8 +81,7 @@ enum RewriteRuleId
   UsuboEliminate,
   SsuboEliminate,
   SdivoEliminate,
-  BVToNatEliminate,
```
Comment (Lachnitt) [inline src/theory/bv/theory_bv_rewrite_rules.h:84]: Why are these deleted?

**Review by Lachnitt — APPROVED** [review]
